### ISSUE TITLE
Prelim Docs Organization

### ISSUE THREAD

**Issue by leeoniya** [leeoniya/domvm#13]
brain dump, for now. pay no attention

```
## Quick Demo

## Philosophy

## Features

## Template Reference

## Views, Models & Composition (domvm.view.js)

## Emit & refs, keys

## Consuming and exposing APIs
	vm.imp
	vm.exp
	vm.redraw(newImp)

## Mutation Observers (domvm.watch.js)

## Routing (domvm.route.js)
	rt.goto
	rt.href
	rt.back/fwd
	rt.state

## Isomorphism (domvm.html.js)
	vm.html()
	vm.attach()

## Third-party DOM integration
	elem			raw elements?
	{_guard: true}	third-party nodes
	{_raw: true}	innerHTML
	vm.absorb()		// also used extreme optimization

## Optimization
	Event Delegation
	Template patching
		vm.patch()

## Useful Patterns

## Solutions to common problems / wiki
	css animation
	animate pre-destroy
	maintain focus
	animated re-order

## Utils (domvm.util.js)

## Suggested Polyfills / es5-shim
	bala.js

## Demos, benchmarks
```

**Comment by leeoniya** [comment]
@barneycarroll
@yosbelms
@lawrence-dol

If any of you have a bit of time and/or interest, I have updated the docs quite a bit and wanted to get some feedback as to whether anything seems unclear, poorly organized or any concepts presented too early or wrong order. I decided to go ahead and add "patterns" from https://github.com/leeoniya/domvm/wiki/View-Model-Patterns to the main readme. Originally I figured it would be overload, but it seems okay and helps with not pushing one opinionated pattern and really shows off the possibilities.

I've also added a bunch of items to the nav.outline that remain on the todo for documentation for features already present in the lib.

thanks!
